### PR TITLE
fix stale props being passed to promiseFn when using watchFn

### LINKED ISSUE #180: Stale data passed to promiseFn when using watchFn
Potentially related #4 

**Issue**
When watching for a variable change using `watchFn`, the react async hook properly re-runs `promiseFn` but with old props passed in.

**Expected**
`promiseFn` should have received the new props on re-run.

**Example**
See https://codesandbox.io/s/jovial-grothendieck-3xth2
In the example above the first render console.logs the following:
```
in function  grapefruit 
in promise grapefruit 
in function  grapefruit 
Rerun promise? 
false
```
All is good! But once we change the fruit, we get the following:
```
in function  lime 
Rerun promise? 
true
in promise grapefruit  <---- should have been lime
in function  lime 
Rerun promise? 
false
```
And if we run it again:
```
in function  coconut 
Rerun promise? 
true
in promise lime    <---- should have been coconut
in function  coconut 
Rerun promise? 
false
```

Am I missing something with the way hooks work or overlooking something simple? Any help will be appreciated!

### PR DESCRIPTION
**Pull request by salolivares**
# Description

When watching for a variable changes using watchFn, the react async hook properly re-runs promiseFn but with old props passed in. This fix aims to correct that behavior.

This fixes #180 

# Checklist

Make sure you check all the boxes. You can omit items that are not applicable.

- [x] Ensure promiseFn receives updated props.
- [x] Added / updated the unit tests

### CODE REVIEW

File: packages/react-async/src/useAsync.js [hunk 1]
```diff
@@ -97,11 +97,11 @@ const useAsync = (arg1, arg2) => {
         .then(handleResolve(counter.current))
         .catch(handleReject(counter.current))
     } else if (promiseFn && !isPreInitialized) {
-      start(() => promiseFn(lastOptions.current, abortController.current))
+      start(() => promiseFn(options, abortController.current))
```
Comment (salolivares) [inline packages/react-async/src/useAsync.js]: Any implications in passing options and adding to the dependency array?

**Comment by Jyrno42** [comment]
Any chance to get this released soon?

**Comment by ghengeveld** [comment]
Thanks for reminding me. We were focused on the TypeScript migration, but that's merged now. This is looking good, so it's going out with the next release, which will be sometime this week.

**Comment by ghengeveld** [comment]
@all-contributors Please add @salolivares for code

**Comment by ghengeveld** [comment]
@all-contributors Please add @salolivares for bug

**Comment by ghengeveld** [comment]
@Jyrno42 I published a pre-release here: [v10.0.0-alpha.0](https://www.npmjs.com/package/react-async/v/10.0.0-alpha.0)

**Pull request merged**